master/buildbot/process/properties.py

```diff
@@ -421,6 +420,33 @@ def getRenderingFor(self, build):
 
 
 @implementer(IRenderable)
+class _SecretRenderer(object):
+
+    def __init__(self, secret_name):
+        self.secret_name = secret_name
+
+    @defer.inlineCallbacks
+    def getRenderingFor(self, build):
+        secretsSrv = build.getBuild().master.namedServices.get("secrets")
+        if not secretsSrv:
+            raise KeyError("secrets service not found")
```

Comment:
need to configure SecretManager in c['services'] to use 'secrets' in Interpolate